grammar.js

```diff
@@ -10,27 +11,30 @@ module.exports = grammar({
     // TODO: add the actual grammar rules
     source_file: $ =>
       seq(
-        $.preamble,
-        repeat($._definition)
+        optional($.control_section),
+        optional($.metadata_section),
+        optional($.shape_section)
```

Comment:
I reorganized the general structure of a source file so it's closer to [`idl` definition from the doc](https://awslabs.github.io/smithy/2.0/spec/idl.html#smithy-idl-abnf)